src/sentry/workflow_engine/handlers/detector/stateful.py

```diff
@@ -279,30 +410,53 @@ def extract_group_values(
         """
         pass
 
-    def build_fingerprint(self, group_key: DetectorGroupKey) -> list[str]:
+    def build_group_fingerprint(self, group_key: DetectorGroupKey) -> list[str]:
         """
         Builds a fingerprint to uniquely identify a detected issue
 
         TODO - Take into account the data source / query that triggered the detector,
         we'll want to create a new issue if the query changes.
         """
-        return [f"{self.detector.id}{':' + group_key if group_key is not None else ''}"]
+        return [self.build_key_for_group(group_key)]
```

Comment:
this could use `self.state_manager._build_key`?

Reply:
+1, i just refactored this so there's only 1 build key method, in the state manager.